src/EditorFeatures/CSharpTest/Diagnostics/DiagnosticAnalyzerDriver/DiagnosticAnalyzerDriverTests.cs

```diff
@@ -226,8 +244,11 @@ void F(int x = 0)
             var analyzer = new CodeBlockAnalyzerFactory();
             using (var ideEngineWorkspace = TestWorkspace.CreateCSharp(source))
             {
+                var analyzerReference = new AnalyzerImageReference(ImmutableArray.Create<DiagnosticAnalyzer>(analyzer));
+                ideEngineWorkspace.TryApplyChanges(ideEngineWorkspace.CurrentSolution.WithAnalyzerReferences(new[] { analyzerReference }));
```

Comment:
Consider adding a helper method in test layer for this common operation, say `void AddHostAnalyzer(this Workspace workspace, DiagnosticAnalyzer analyzer)`

Reply:
I'll follow up on that in a separate PR.